pkg/apis/application/v1alpha1/types.go

```diff
@@ -919,6 +924,8 @@ func (m *Repository) CopyCredentialsFrom(source *Repository) {
 		m.InsecureIgnoreHostKey = source.InsecureIgnoreHostKey
 		m.Insecure = source.Insecure
 		m.EnableLFS = source.EnableLFS
```

Comment:
is this really credential related

Reply:
No, it isn't - and I feel all this `Creds` thing could need a refactor of meaning. The `Creds` struct is something currently passed from API server to repo server,  and it should imho contain all information required for correctly connecting to and accessing the repository. So, I'd like to start a discussion about changing it from something along "credential" to something more generic such as "repository access parameters". Opinions on that?

Reply:
this func is used to copy the credentials from another "repo credential" (typically these apply to many repos) - so I think you should probably not have it here

Reply:
I think this is not quite true, and the relationship between repo and credential set is `1:1` currently. As far as my research went, `CopyCredentialsFrom` is used only to copy the credential sets from API server to repo server, so that repo server knows how to properly connect to the repository. But as I said, I wouldn't be opposed to refactoring that whole credentials vs. connection properties thing, but I think it will require some serious thought and also quite a bit of work.

Reply:
can I point you to this please:

https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/#repository-credentials

Reply:
Hm. OK, that's something I did not consider initially. 

However, as far as I understand the code, it works like that in the repository setup phase or on a change event:

- Get repository definiton from ConfigMap
- If repo definition has credentials defined, use them and setup `RepoCredentials`
- If repo definition don't have credentials defined, use definition from `repository.credentials` (if exist) and populate defined fields in `RepoCredentials` for the repository.

It does not look to me that the repo actually uses a pointer to the repository.credentials for `RepoCredentials`, so that's why I say `1:1` relationship (at least, down in the code).

So I *think* that upon change of credentials in `repository.credentials`, not a complete `RepoCredentials` struct would be set on the repository (i.e. using `CopyCredentialsFrom`), but just the changed values such as username/password.

Considering all this, my approach sounds a little fragile tho. So maybe, we should have a `RepoOptions` data structure holding stuff like `Insecure` and `EnableLFS` which is then passed down in the code stream.

How shall we proceed?

Reply:
I think just delete `m.EnableLFS = source.EnableLFS`.

Reply:
That would definately break LFS support, because the repo-server needs this information and currently, `CopyCredentialsFrom` is how it gets this information. Need to rework some code for that (oh, and add LFS to E2E tests, too).

Reply:
ok, this is unrelated to this PR